File: python/7 kyu/113.py
# https://www.codewars.com/kata/58b972cae826b960a300003e/train/python

# Passed

def missing(nums, s):
    joined_code = "".join(s.split())
    deciphered_code = ""
    nums.sort()
    for idx in nums:
        if idx >= len(joined_code):
            return "No mission today"

        deciphered_code += joined_code[idx]
    
    deciphered_code = deciphered_code.lower()
    return deciphered_code
         
output = missing([29, 31, 8], "The quick brown fox jumps over the lazy dog")
print(output)

File: nodejs/6 kyu/113.js
// https://www.codewars.com/kata/5897cdc26551af891c000124/train/javascript

// Passed

function hofstadterQ(n, memo = {}) {
  if (n === 1 || n === 2) return 1;

  if (n in memo) {
    return memo[n];
  }

  // caching to memo
  let result =
    hofstadterQ(n - hofstadterQ(n - 1, (memo = memo)), (memo = memo)) +
    hofstadterQ(n - hofstadterQ(n - 2, (memo = memo)), (memo = memo));
  memo[n] = result;

  return result;
}

const output = hofstadterQ(1000);
console.log(output);